src/paths/astar.c

```diff
@@ -0,0 +1,418 @@
+/*
+   IGraph library.
+   Copyright (C) 2022 The igraph development team
+
+   This program is free software; you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation; either version 2 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program; if not, write to the Free Software
+   Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+   02110-1301 USA
+
+*/
+
+#include "igraph_paths.h"
+#include "igraph_interface.h"
+#include "igraph_vector_list.h"
+#include "igraph_adjlist.h"
+#include "igraph_memory.h"
+
+#include "core/indheap.h"
+#include "core/interruption.h"
+
+/**
+ * \ingroup structural
+ * \function igraph_get_shortest_paths_astar
+ * \brief Shortest paths from a vertex with heuristic.
+ *
```

Comment:
I suggest to make it a policy to mark all non-trivial functions added in non-major versions (i.e. mid-cycle between 0.10.0 and 0.11.0) as `\experimental`. Then most these `\experimental` labels can be removed in 0.11.0.